src/message.rs

```diff
@@ -1694,38 +1740,20 @@ pub async fn delete_msgs(context: &Context, msg_ids: &[MsgId]) -> Result<()> {
             res = Err(e);
             continue;
         }
-
-        let logging_xdc_id = context
-            .debug_logging
-            .read()
-            .expect("RwLock is poisoned")
-            .as_ref()
-            .map(|dl| dl.msg_id);
-
-        if let Some(id) = logging_xdc_id {
-            if id == msg_id {
-                set_debug_logging_xdc(context, None).await?;
-            }
-        }
     }
     res?;
 
-    for modified_chat_id in modified_chat_ids {
-        context.emit_msgs_changed_without_msg_id(modified_chat_id);
-        chatlist_events::emit_chatlist_item_changed(context, modified_chat_id);
-    }
-
-    if !msg_ids.is_empty() {
-        context.emit_msgs_changed_without_ids();
-        chatlist_events::emit_chatlist_changed(context);
-        // Run housekeeping to delete unused blobs.
-        context
-            .set_config_internal(Config::LastHousekeeping, None)
-            .await?;
-    }
+    delete_msgs_locally_done(context, msg_ids, modified_chat_ids).await?;
 
     // Interrupt Inbox loop to start message deletion and run housekeeping.
     context.scheduler.interrupt_inbox().await;
+
+    context
+        .add_sync_item(SyncData::DeleteMessages {
+            msgs: deleted_rfc724_mid,
+        })
+        .await?;
+    context.send_sync_msg().await?;
```

Comment:
See the documentation for this function, you should call `context.scheduler.interrupt_inbox()` instead. This should be fixed in `chat::save_msgs()` as well btw

Reply:
i added a commit as you suggested, but i am not sure if i am doing it correctly - so `context.scheduler.interrupt_inbox()` satisfies the "We guarantee this by calling `send_sync_msg()` only from the SMTP loop."  constraint?

Reply:
> i added a commit as you suggested, but i am not sure if i am doing it correctly - so `context.scheduler.interrupt_inbox()` satisfies the "We guarantee this by calling `send_sync_msg()` only from the SMTP loop." constraint?

Yes. An attempt to quickly clarify this:

The comment is actually wrong, we're not calling it from the SMTP loop but from the inbox loop.

`interrupt_inbox()` makes the imap loop interrupt the idle, do a lot of work including `send_sync_msg()`, and go into idle again. `send_sync_msg()` will prepare the message and then interrupt the SMPT loop if necessary, so that it's actually sent out.

Reply:
The comment should  be fixed of course, probably it's stale. Moreover, for sync messages the SMTP loop doesn't need to do anything, they are uploaded using `IMAP APPEND` by the inbox loop.